File: Magick.NET/Drawables/DrawableAffine.cpp
//=================================================================================================
// Copyright 2013-2014 Dirk Lemstra <https://magick.codeplex.com/>
//
// Licensed under the ImageMagick License (the "License"); you may not use this file except in 
// compliance with the License. You may obtain a copy of the License at
//
//   http://www.imagemagick.org/script/license.php
//
// Unless required by applicable law or agreed to in writing, software distributed under the
// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
// express or implied. See the License for the specific language governing permissions and
// limitations under the License.
//=================================================================================================
#include "Stdafx.h"
#include "DrawableAffine.h"

namespace ImageMagick
{
	//==============================================================================================
	DrawableAffine::DrawableAffine(double scaleX, double scaleY, double shearX, double shearY,
		double translateX, double translateY)
	{
		BaseValue = new Magick::DrawableAffine(scaleX, scaleY, shearX, shearY, translateX, translateY);
	}
	//==============================================================================================
	DrawableAffine::DrawableAffine(Matrix^ matrix)
	{
		Throw::IfNull("matrix", matrix);

		BaseValue = new Magick::DrawableAffine((double)matrix->Elements[0], (double)matrix->Elements[1],
			(double)matrix->Elements[2], (double)matrix->Elements[3], (double)matrix->Elements[4],
			(double)matrix->Elements[5]);
	}
	//==============================================================================================
	double DrawableAffine::ScaleX::get()
	{
		return Value->sx();
	}
	//==============================================================================================
	void DrawableAffine::ScaleX::set(double value)
	{
		Value->sx(value);
	}
	//==============================================================================================
	double DrawableAffine::ScaleY::get()
	{
		return Value->sy();
	}
	//==============================================================================================
	void DrawableAffine::ScaleY::set(double value)
	{
		Value->sy(value);
	}
	//==============================================================================================
	double DrawableAffine::ShearX::get()
	{
		return Value->rx();
	}
	//==============================================================================================
	void DrawableAffine::ShearX::set(double value)
	{
		Value->rx(value);
	}
	//==============================================================================================
	double DrawableAffine::ShearY::get()
	{
		return Value->ry();
	}
	//==============================================================================================
	void DrawableAffine::ShearY::set(double value)
	{
		Value->ry(value);
	}
	//==============================================================================================
	double DrawableAffine::TranslateX::get()
	{
		return Value->tx();
	}
	//==============================================================================================
	void DrawableAffine::TranslateX::set(double value)
	{
		Value->tx(value);
	}
	//==============================================================================================
	double DrawableAffine::TranslateY::get()
	{
		return Value->ty();
	}
	//==============================================================================================
	void DrawableAffine::TranslateY::set(double value)
	{
		Value->ty(value);
	}
	//==============================================================================================
}

File: Magick.NET/Drawables/DrawableAffine.cs
//=================================================================================================
// Copyright 2013-2016 Dirk Lemstra <https://magick.codeplex.com/>
//
// Licensed under the ImageMagick License (the "License"); you may not use this file except in 
// compliance with the License. You may obtain a copy of the License at
//
//   http://www.imagemagick.org/script/license.php
//
// Unless required by applicable law or agreed to in writing, software distributed under the
// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
// express or implied. See the License for the specific language governing permissions and
// limitations under the License.
//=================================================================================================

using System.Drawing.Drawing2D;
using ImageMagick.Drawables;

namespace ImageMagick
{
  ///<summary>
  /// Adjusts the current affine transformation matrix with the specified affine transformation
  /// matrix. Note that the current affine transform is adjusted rather than replaced.
  ///</summary>
  public sealed class DrawableAffine : IDrawable
  {
    void IDrawable.Draw(IDrawingWand wand)
    {
      if (wand != null)
        wand.Affine(ScaleX, ScaleY, ShearX, ShearY, TranslateX, TranslateY);
    }

    ///<summary>
    /// Creates a new DrawableAffine instance.
    ///</summary>
    ///<param name="scaleX">The X coordinate scaling element.</param>
    ///<param name="scaleY">The Y coordinate scaling element.</param>
    ///<param name="shearX">The X coordinate shearing element.</param>
    ///<param name="shearY">The Y coordinate shearing element.</param>
    ///<param name="translateX">The X coordinate of the translation element.</param>
    ///<param name="translateY">The Y coordinate of the translation element.</param>
    public DrawableAffine(double scaleX, double scaleY, double shearX, double shearY, double translateX, double translateY)
    {
      ScaleX = scaleX;
      ScaleY = scaleY;
      ShearX = shearX;
      ShearY = shearY;
      TranslateX = translateX;
      TranslateY = translateY;
    }

    ///<summary>
    /// Creates a new DrawableAffine instance using the specified Matrix.
    ///</summary>
    ///<param name="matrix">The matrix.</param>
    public DrawableAffine(Matrix matrix)
    {
      Throw.IfNull("matrix", matrix);

      ScaleX = matrix.Elements[0];
      ScaleY = matrix.Elements[1];
      ShearX = matrix.Elements[2];
      ShearY = matrix.Elements[3];
      TranslateX = matrix.Elements[4];
      TranslateY = matrix.Elements[5];
    }

    ///<summary>
    /// The X coordinate scaling element.
    ///</summary>
    public double ScaleX
    {
      get;
      set;
    }

    ///<summary>
    /// The Y coordinate scaling element.
    ///</summary>
    public double ScaleY
    {
      get;
      set;
    }

    ///<summary>
    /// The X coordinate shearing element.
    ///</summary>
    public double ShearX
    {
      get;
      set;
    }

    ///<summary>
    /// The Y coordinate shearing element.
    ///</summary>
    public double ShearY
    {
      get;
      set;
    }

    ///<summary>
    /// The X coordinate of the translation element.
    ///</summary>
    public double TranslateX
    {
      get;
      set;
    }

    ///<summary>
    /// The Y coordinate of the translation element.
    ///</summary>
    public double TranslateY
    {
      get;
      set;
    }
  }
}